numpy/core/arrayprint.py

```diff
@@ -607,14 +600,23 @@ def _formatArray(a, format_function, rank, max_line_len,
 
 class FloatFormat(object):
     def __init__(self, data, precision, suppress_small, sign=False):
+        # for backcompatibility, accept bools
+        if sign is False:
+            sign = '-'
+        if sign is True:
+            sign = '+'
```

Comment:
Not sure about using `is` on booleans. How about

```
if isinstance(sign, bool):
    sign = '+' if sign else '-'
```

Reply:
On the sign, see https://github.com/numpy/numpy/pull/9139#issuecomment-312456857 - I think we should follow precedent and let it be one of `" ", "+", "-"`

Reply:
We already do - but we need to support booleans here for backwrads compatibility